ts/Core/Chart/Chart.ts

```diff
@@ -622,28 +622,87 @@ class Chart {
      * @param {number} plotY
      * Pixel y relative to the plot area.
      *
-     * @param {boolean} [inverted]
-     * Whether the chart is inverted.
+     * @param {Highcharts.ChartIsInsideOptionsObject} [options]
+     * Options object.
      *
      * @return {boolean}
      * Returns true if the given point is inside the plot area.
      */
     public isInsidePlot(
         plotX: number,
         plotY: number,
-        inverted?: boolean
+        options: Chart.IsInsideOptionsObject = {}
     ): boolean {
-        var x = inverted ? plotY : plotX,
-            y = inverted ? plotX : plotY,
-            e = {
-                x,
-                y,
-                isInsidePlot: x >= 0 &&
-                    x <= this.plotWidth &&
-                    y >= 0 &&
-                    y <= this.plotHeight
+        const {
+            inverted,
+            plotBox,
+            plotLeft,
+            plotTop,
+            scrollablePlotBox,
+            scrollingContainer: {
+                scrollLeft,
+                scrollTop
+            } = {
+                scrollLeft: 0,
+                scrollTop: 0
+            }
+        } = this;
+
+        const series = options.series;
+        const box = (options.visiblePlotOnly && scrollablePlotBox) || plotBox;
```

Comment:
It seems that the new `visiblePlotOnly` property is passed frequently. Is it more often true or more often false? If it is more often `true`, maybe we can make `true` the default? It would save bytes and be more backwards compatible?

Reply:
About 12 out of 34 calls pass it.